### PR TITLE
Remove `lib`+`.so` from plugin's name

### LINKED ISSUE #277: Remove lib+.so from plugin names on examples
One of the nice features of `ign-plugin` compared to Gazebo-classic's plugin loader is that it supports passing the library name without the `lib` prefix and the `.so` extension. This is nice because the extension is not `.so` on all platforms.

Currently, our examples make extensive use of `.so` though, i.e.:

https://github.com/ignitionrobotics/ign-gazebo/blob/8ec8d66cc8a6b661a2db6ccae153a785287fa315/examples/worlds/empty.sdf#L60-L75

These could be:

```
     <plugin
       filename="ignition-gazebo-physics-system"
       name="ignition::gazebo::systems::Physics">
     </plugin>
     <plugin
       filename="ignition-gazebo-user-commands-system"
       name="ignition::gazebo::systems::UserCommands">
     </plugin>
     <plugin
       filename="ignition-gazebo-scene-broadcaster-system"
       name="ignition::gazebo::systems::SceneBroadcaster">
     </plugin>
```

Since that's the cleanest and generic way of defining the worlds, we should update the examples.

### PR DESCRIPTION
**Pull request by mohamedsayed18**
Fix #277.
Is this PR big and need to be broken down into smaller ones?

### CODE REVIEW

**Comment by mohamedsayed18** [comment]
> It doesn't look like you consistently remove the `.so` extension from the plugin filenames. Is there a reason for this?

No, it was just a mistake.
I searched for the `.so` and I found these two files:
* ./build/ignition-gazebo3/CMakeCache.txt:325 `//Install .so files without execute permission.`
* tutorials/physics.md:97 `If that's a 3rd party engine, find where the .so file is installed and add`
Should I remove those too?

File: examples/scripts/distributed/primary.sdf [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <sdf version="1.6">
   <world name="default">
     <plugin
-      filename="libignition-gazebo-scene-broadcaster-system.so"
+      filename="ignition-gazebo-scene-broadcaster-system.so"
```
Comment (chapulina) [inline examples/scripts/distributed/primary.sdf:5]: @mohamedsayed18 , some example files still have `.so`

**Comment by mohamedsayed18** [comment]
I think I shouldn't remove it from the custom plugins like `libCustomEngine.so`, `libLiftDragPlugin.so`, `libSampleSystem.so`?

Should I remove it from the `ign-gazebo/test` directory?

**Comment by chapulina** [comment]
> I think I shouldn't remove it from the custom plugins 

It can be removed from all of then, `ign-plugin` will be able to find them :wink: 

> Should I remove it from the ign-gazebo/test directory?

That would be great, thanks!

**Comment by mohamedsayed18** [comment]
I will work on it.

File: examples/scripts/distributed_levels/distributed_levels.sdf.erb [hunk 2]
```diff
@@ -103,14 +103,14 @@
 
     <% if secondary %>
     <plugin
-     filename="libignition-gazebo-physics-system.so"
+     filename="libignition-gazebo-physics-system"
```
Comment (mjcarroll) [inline examples/scripts/distributed_levels/distributed_levels.sdf.erb:106]: For example, here you don't drop the `lib` prefix.

File: examples/scripts/distributed_levels/primary.sdf [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 
     <plugin
-      filename="libignition-gazebo-scene-broadcaster-system.so"
+      filename="ignition-gazebo-scene-broadcaster-system"
```
Comment (mjcarroll) [inline examples/scripts/distributed_levels/primary.sdf:17]: But both `lib` and `.so` are removed here.

**Comment by mohamedsayed18** [comment]
I didn't remove the extension name from the `src/` and the `test/integration`

**Pull request merged**